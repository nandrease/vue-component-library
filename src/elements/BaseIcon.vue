<template>
    <svg class="icon" :class="`icon--${size}`" :style="styles">
        <use :href="`${iconsUrl}#${icon}`"></use>
    </svg>
</template>

<script>
export default {
    props: {
        icon: {
            type: String,
            default: ''
        },
        fill: {
            type: String
        }
    },
    computed: {
        size() {
            const num = this.icon.split('-').pop();
            return isNaN(parseInt(num)) ? 16 : parseInt(num);
        },
        styles() {
            return this.fill ? { fill: this.fill } : null;
        },
        iconsUrl() {
            return process.env.VUE_APP_ICONS_URL || '';
        }
    }
};
</script>

<style lang="scss"></style>
