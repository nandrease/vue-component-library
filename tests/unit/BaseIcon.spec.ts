import { shallowMount, Wrapper } from "@vue/test-utils";
import BaseIcon from "@/elements/BaseIcon.vue";

let wrapper: Wrapper<Vue>;

const propsData = {
  icon: "icon-search-16"
};

beforeEach(() => (wrapper = shallowMount(BaseIcon, { propsData })));
afterEach(() => wrapper.destroy());

describe("<ScoroIcon />", () => {
  it("renders and displays properly", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has a use element with defined icon name and size", () => {
    const useElem = wrapper.find("use");
    expect(useElem.attributes("href")).toContain(propsData.icon);
  });

  it("prperly detects size from icon and uses it as a class on wrapper", () => {
    expect(wrapper.classes()).toContain("icon--16");
    wrapper = shallowMount(BaseIcon, {
      propsData: {
        icon: "icon-search-24"
      }
    });
    expect(wrapper.classes()).toContain("icon--24");
  });

  it("it uses fill prop to color the element", () => {
    expect(wrapper.attributes().style).toBeFalsy();
    wrapper = shallowMount(BaseIcon, {
      propsData: {
        ...propsData,
        fill: "rebeccapurple"
      }
    });
    expect(wrapper.attributes().style).toBe("fill: rebeccapurple;");
  });
});
