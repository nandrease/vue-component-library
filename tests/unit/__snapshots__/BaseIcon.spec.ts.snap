// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoroIcon /> renders and displays properly 1`] = `
<svg
  class="icon icon--16"
>
  <use
    href="#icon-search-16"
  />
</svg>
`;
